src/main/java/joliex/db/DBConnector.java

```diff
@@ -1,36 +1,603 @@
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this program; if not, write to the                 *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ *                                                                         *
+ *   For details about the authors of this software, see the AUTHORS file. *
+ ***************************************************************************/
+
 package joliex.db;
 
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.math.BigDecimal;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import jolie.runtime.ByteArray;
+import jolie.runtime.CanUseJars;
+import jolie.runtime.FaultException;
 import jolie.runtime.JavaService;
+import jolie.runtime.Value;
+import jolie.runtime.ValueVector;
 import jolie.runtime.embedding.RequestResponse;
+import com.mchange.v2.log.*;
+import com.mchange.v2.c3p0.DataSources;
+import java.util.logging.Level;
+import javax.sql.DataSource;
+
 
 /**
- * A {@link JavaService} for managing databases.
- * 
- * @author Fabrizio Montesi
+ * @author Fabrizio Montesi 2008
+ * - Marco Montesi: connection string fix for Microsoft SQL Server (2009)
+ * - Claudio Guidi: added support for SQLite (2013)
+ * - Matthias Dieter Wallnöfer: added support for HSQLDB (2013)
  */
+@CanUseJars( {
+	"derby.jar", // Java DB - Embedded
+	"derbyclient.jar", // Java DB - Client
+	"jdbc-mysql.jar", // MySQL
+	"jdbc-postgresql.jar", // PostgreSQL
+	"jdbc-sqlserver.jar", // Microsoft SQLServer
+	"jdbc-sqlite.jar", // SQLite
+	"jt400.jar", // AS400
+	"hsqldb.jar", // HSQLDB
+	"db2jcc.jar" // DB2
+} )
 public class DBConnector extends JavaService
 {
+	//private Connection connection = null;
+	private String connectionString = null;
+	private String username = null;
+	private String password = null;
+	private String driver = null;
+	private static boolean toLowerCase = false;
+	private static boolean toUpperCase = false;
+	private final Object transactionMutex = new Object();
+	private final static String templateField = "_template";
+	DataSource ds_pooled = null;
+	
```

Comment:
If we make a type just for the config parameters (see one of my comments on the .iol), we should consider using a ValueConverter (see ReplaceAllRequest in StringUtils).